templates/fine-tune-llm_v2/end-to-end-examples/fine-tune-preference/README.ipynb

```diff
@@ -0,0 +1,1477 @@
+      "Q4) Why did Jennifer Lawrence's mom take her to New York at the age of 14? A. To\n",
+      "visit relatives B. To attend a sports tournament C. To go shopping D. For\n",
+      "auditions E. For a family vacation  Q5) What was Jennifer Lawrence's role in the\n",
+      "movie \"Winter's Bone\"? A. A character in a Jodie Foster movie B. A mutant with\n",
+      "blue skin C. A character in a Reese's Peanut Butter Cups commercial D. A\n",
+      "frightened waif E. A 17-year-old daughter of an Ozarks meth-cooker\n",
+      "\n",
+      "MODEL GENERATED_SUMMARY:\n",
+      "Jennifer Lawrence, an Oscar-nominated actress from Kentucky, reflects on her\n",
+      "journey from a tomboy to a successful actress in Hollywood. She talks about her\n",
+      "love for field hockey, softball, and basketball, and her nickname \"Nitro.\"\n",
+      "Lawrence auditioned for the role of Bella in \"Twilight\" but didn't get it, but\n",
+      "went on to star in \"Winter's Bone,\" where she received critical acclaim. She\n",
+      "discusses her preparation for the role, including learning how to shoot a gun\n",
+      "and field-dress squirrels. Lawrence also talks about her upcoming roles in\n",
+      "\"X-Men: First Class\" and \"The Beaver,\" and shares some crazy Mel Gibson stories.\n",
+      "\n",
+      "ANSWERS:\n",
+      "['E' 'B' 'E' 'D' 'E']\n",
+      "\n",
+      "JUDGE ANSWERS FROM SUMMARY:\n",
+      "['E', 'B', 'E', 'Unsure', 'Unsure']\n",
+      "\n",
+      "'===================================================================================================='\n"
+     ]
+    }
+   ],
+   "source": [
+    "from src.utils.models import DataSchema\n",
+    "\n",
+    "for row in example_rows:\n",
+    "    print_wrapped(\"TEXT\", row[DataSchema.ARTICLE])\n",
+    "    print_wrapped(\"QUESTIONS\", row[DataSchema.MCQ_QUESTIONS])\n",
+    "    print_wrapped(\"MODEL GENERATED SUMMARY\", row[DataSchema.SUMMARY_GENERATION_RAW_OUTPUT])\n",
+    "    print_wrapped(\"ANSWERS\", str(row[DataSchema.GROUND_TRUTH_MCQ_ANSWERS]))\n",
+    "    print_wrapped(\"JUDGE ANSWERS FROM SUMMARY\", str(row[DataSchema.JUDGE_MCQ_ANSWERS]))\n",
+    "    pprint.pprint(\"=\" * 100, width=80)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Part (c): Generate Preference Tuning Data\n",
+    "\n",
+    "Next, we will generate 10 summaries for each article in the training set and score them with our Q&A judging setup. \n",
+    "\n",
+    "The following command will run the [generate_dpo_data.py](src/scripts/generate_dpo_data.py) script, which takes in the folder of summaries and outputs `.jsonl` files for training and validation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!python src/scripts/generate_dpo_data.py configs/training_data_generation/mistral_8b.yaml"
```

Comment:
This should be also a job submit?

Reply:
Why? It just performs a bunch of `ds.map` operations? I think the fine-tuning template can handle this?

I want to simplify script execution wherever possible.  This script and the `get_eval_stats` script I think should run locally to get the results right away. 


Wdyt?

Reply:
ah ok I wasn't sure how long it takes. I want the user implicetly get into following a pattern of if a script takes long / needs different dependencies (aica image vs. llmforge) or need a lot of gpus just launch a job and continue with developmenet on workspace without interruption. so can we make sure we don't have users run stuff within the notebooks? like never do `!python ...` have them run the cli from the terminal. It will also help keep the notebook clean from randomly populated output cells.